src/main/java/org/jboss/forge/arquillian/container/provider/ContainerDirectoryParser.java

```diff
@@ -35,15 +40,23 @@ void createContainers() {
                 new TypeReference<List<Container>>() {
                 });
 
-            containers = Collections.unmodifiableList(parsedContainers);
+            Loader loader = new Loader();
+            final org.arquillian.container.chameleon.spi.model.Container[] containers = loader.loadContainers(Target.class.getClassLoader().getResourceAsStream("chameleon/default/containers.yaml"));
+
+            parsedContainers.stream()
+                .filter(container -> container.getGroupId() == null && container.getArtifactId() == null)
+                .forEach(container -> {
+                    try {
+                        container.setGroupIdAndArtifactIdFromChameleonConfiguration(containers);
```

Comment:
This piece could be improved I believe. We do the same looping over `containers` for each and every instance of `parsedContainer`. Maybe this logic shouldn't belong there at all? Can you try to decouple the logic in `setGroupIdAndArtifactIdFromChameleonConfiguration` to its own component and handle mapping at once for all `parsedContainers`?